### PR TITLE
#486 - work package creation fails if project has no work packages

### PR DESCRIPTION
**Pull request by anthonybernardi**
I think I fixed the bug but I can't run it to check because I'm having issues with my dev environment

### CODE REVIEW

**Comment by acd124** [comment]
There's also a bug where it's finding the max work package number for all projects with said project number across all cars, so if there's a project 1 on car 1, and a project 1 on car 2, it will try to find the max of both's work package numbers. I think the cleanest solution is to not query the database again and simply use the work packages provided in `project.workPackages` and reduce it to the max workPackageId

File: src/backend/functions/work-packages-create.ts [hunk 1]
```diff
@@ -38,16 +38,20 @@ export const createWorkPackage: Handler = async ({ body }, _context) => {
...
 
   if (project === null) throw new TypeError('Project Id not found!');
 
   // eslint-disable-next-line prefer-destructuring
-  const { carNumber, projectNumber } = project.workPackages[0].wbsElement;
+  const { carNumber, projectNumber } = project.wbsElement;
 
-  const workPackageNumber = (await getHighestWorkPackageNumber(projectNumber)) + 1;
+  const workPackageNumber =
+    project.workPackages === undefined || project.workPackages.length === 0
```
Comment (acd124) [inline src/backend/functions/work-packages-create.ts:52]: If you check the types, you'll see that `project.workPackages` has a type of something. like `Work_Packages & {...}` based on the prisma query, indicating that it cannot be undefined. if it ever is undefined we have prisma to blame for lying to us, so there's no need to check.

Comment (anthonybernardi) [reply src/backend/functions/work-packages-create.ts:52]: gotcha, makes sense

File: src/backend/functions/work-packages-create.ts [hunk 2]
```diff
@@ -38,16 +24,22 @@ export const createWorkPackage: Handler = async ({ body }, _context) => {
       goals: true,
       features: true,
       otherConstraints: true,
```
Comment (acd124) [inline src/backend/functions/work-packages-create.ts:26]: Not sure why these other properties are being asked for, I know you didn't touch them, but I believe these 3 can be removed without any effect.

**Review by acd124 — DISMISSED** [review]
The bug looks fixed, and fewer requests are being made to the database. Great work, looks good to me 👍🏼

File: src/backend/functions/work-packages-create.ts (hunk 2 above)
Comment (anthonybernardi) [reply src/backend/functions/work-packages-create.ts:26]: true, good point

**Review by jamescd18 — APPROVED** [review]
Much cleaner, I love it!

**Pull request merged**